go/vt/topo/keyspace.go

```diff
@@ -270,26 +273,74 @@ func (ts *Server) UpdateKeyspace(ctx context.Context, ki *KeyspaceInfo) error {
 	return nil
 }
 
-// FindAllShardsInKeyspace reads and returns all the existing shards in
-// a keyspace. It doesn't take any lock.
-func (ts *Server) FindAllShardsInKeyspace(ctx context.Context, keyspace string) (map[string]*ShardInfo, error) {
+// FindAllShardsInKeyspaceConfig controls the behavior of
+// Server.FindAllShardsInKeyspace.
+type FindAllShardsInKeyspaceConfig struct {
+	// Concurrency controls the maximum number of concurrent calls to GetShard.
+	// If unspecified, Concurrency is set to 1.
+	Concurrency int
+}
+
+// FindAllShardsInKeyspace reads and returns all the existing shards in a
+// keyspace. It doesn't take any lock.
+//
+// If cfg is non-nil, it is used to configure the method's behavior. Otherwise,
+// a default configuration is used.
+func (ts *Server) FindAllShardsInKeyspace(ctx context.Context, keyspace string, cfg *FindAllShardsInKeyspaceConfig) (map[string]*ShardInfo, error) {
+	if cfg == nil || cfg.Concurrency == 0 {
```

Comment:
Will break this into two conditionals as a follow up, don't want to accidentally overwrite future config options if this one is unset.